mono/metadata/image.c

```diff
@@ -2538,49 +2538,24 @@ mono_image_append_class_to_reflection_info_set (MonoClass *klass)
 
 #if CHECKED_BUILD
 
-// These are support for the mempool reference tracking feature in checked-build, but live in image.c due to use of static variables of this file.
-
-// Given an image and a pointer, return the mempool owner if it is either this image or one of its imagesets.
-static MonoMemPoolOwner
-check_for_mempool_owner (void *ptr, MonoImage *image)
-{
-	if (mono_mempool_contains_addr (image->mempool, ptr))
-	{
-		MonoMemPoolOwner owner = {image, NULL};
-		return owner;
-	}
-
-	GSList *l;
-	for (l = image->image_sets; l; l = l->next) {
-		MonoImageSet *set = l->data;
-
-		if (mono_mempool_contains_addr (set->mempool, ptr))
-		{
-			MonoMemPoolOwner owner = {NULL, set};
-			return owner;
-		}
-	}
-
-	return mono_mempool_no_owner;
-}
+// This is support for the mempool reference tracking feature in checked-build, but lives in image.c due to use of static variables of this file.
 
 /**
- * mono_find_mempool_owner:
+ * mono_find_image_owner:
  *
- * Find the image or imageset, if any, which a given pointer is located in the memory of.
+ * Find the disk-loaded image, if any, which a given pointer is located in the memory of.
```

Comment:
mages don't need to be disk-loaded, there are other ways to load them besides disk files that will in turn produce non-dynamic images. mkbundle'd assemblies and Assembly.Load (byte[]) are two examples of that.